src/test/java/hw4/base/TestListener.java

```diff
@@ -0,0 +1,30 @@
+package hw4.base;
+
+import io.qameta.allure.Attachment;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.WebDriver;
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+
+public class TestListener implements ITestListener {
+
+    @Attachment(value = "Page screenshot", type = "image/png")
+    public byte[] saveScreenShot(WebDriver driver) {
+        return ((TakesScreenshot) driver).getScreenshotAs(OutputType.BYTES);
+    }
+
+    private static String getTestMethodName(ITestResult itestResult) {
+        return itestResult.getMethod().getConstructorOrMethod().getName();
+    }
+
+    @Override
+    public void onTestFailure(ITestResult result) {
+        WebDriver driver = ((BaseTests)result.getInstance()).getDriver();
+        if (driver instanceof WebDriver) {
```

Comment:
If execution flow passes through this line then it is guaranteed that driver is an instance of WebDriver, so you can only compare with null

Reply:
fixed